packages/gatsby-plugin-google-analytics/src/gatsby-ssr.js

```diff
@@ -1,17 +1,29 @@
 import React from "react"
 
 const knownOptions = {
-  clientId: `string`,
-  sampleRate: `number`,
-  siteSpeedSampleRate: `number`,
-  alwaysSendReferrer: `boolean`,
-  allowAnchor: `boolean`,
-  cookieName: `string`,
-  cookieExpires: `number`,
-  storeGac: `boolean`,
-  legacyCookieDomain: `string`,
-  legacyHistoryImport: `boolean`,
-  allowLinker: `boolean`,
+  createOnly: {
+    clientId: `string`,
+    sampleRate: `number`,
+    siteSpeedSampleRate: `number`,
+    alwaysSendReferrer: `boolean`,
+    allowAnchor: `boolean`,
+    cookieName: `string`,
+    cookieExpires: `number`,
+    storeGac: `boolean`,
+    legacyCookieDomain: `string`,
+    legacyHistoryImport: `boolean`,
+    allowLinker: `boolean`,
+  },
+  general: {
+    allowAdFeatures: `boolean`,
+    dataSource: `string`,
+    queueTime: `number`,
+    forceSSL: `boolean`,
+    transport: `string`,
+  },
+  aLaCarte: {
```

Comment:
could you elaborate on why this is added? I can't seem to find this being used in your changes?

Reply:
Absolutely! Good question 👍 

The `aLaCarte` section is for options that are not directly from the Google Analytics API/docs, but rather options specific to this Gatsby Plugin. 
The context is that the `anonymize` option is not a Google Analytics option (not to be confused with the very similar GA-official option `anonymizeIp`).

Upon further contemplation, the name is kind of vague. Thus I will be renaming the section to `pluginSpecific` and adding a comment to clarify what the section is for. 🔥